client/ayon_usd/standalone/usd/pinning/pinning_file_helper.py

```diff
@@ -0,0 +1,280 @@
+import json
+import os
+import re
+from typing import Dict, List
+from pxr import UsdShade, Ar, Sdf
+from urllib.parse import urlparse
+
+
+def is_uri(path: str) -> bool:
+    parsed = urlparse(path)
+    return bool(parsed.scheme)
+
+
+# Assume that the environment sets up the correct default AyonUsdResolver
+resolver = Ar.GetResolver()
+
+
+def remove_root_from_dependency_info(
+    dependency_info: Dict[str, str], root_info: Dict[str, str]
+) -> Dict[str, str]:
+    """Re-roots a path dict so that a Ayon_Usd_Resolver can use its own system for root replace to resolve a given path
+
+    Args:
+        dependecy_info: dict generated by get_asset_dependencies()
+        root_info: a flat dict containing str: root identifier {root[work]}, str: associated root path /path/to/root
+
+    Returns: the input dependecy_info dict with replaced root_info to obtain root-less paths,
+    """
+
+    if not root_info or not dependency_info:
+        return {}
+
+    replacements = {
+        re.escape(path_portion): replacer
+        for replacer, path_portion in root_info.items()
+    }
+
+    pattern = re.compile("|".join(f"({pat})" for pat in replacements))
+
+    def _replace_match(match: re.Match):
+        for _, group in enumerate(match.groups(), start=1):
+            if group:
+                replacment = "{root[" + replacements[re.escape(group)] + "]}"
+                return replacment
+        return match.group(0)
+
+    root_less_dict = {}
+
+    for key, path in dependency_info.items():
+        new_path = pattern.sub(_replace_match, path)
+        root_less_dict[key] = new_path
+
+    return root_less_dict
+
+
+def _get_prim_prop_data(prim: Sdf.PrimSpec, layer: Sdf.Layer) -> List[str]:
+    prop_data = []
+    for prop in prim.properties:
+        if isinstance(prop, Sdf.AttributeSpec):
+            if prop.typeName == Sdf.ValueTypeNames.Asset:
+
+                default_val = prop.default
+                if default_val:
+                    if hasattr(default_val, "__iter__"):
+                        prop_data.extend(default_val)
+                    else:
+                        prop_data.append(default_val.path)
+                    continue  # No need to check for time samples if default val is populated
+
+                time_samples = layer.ListTimeSamplesForPath(prop.path)
+                for time in time_samples:
+                    value = layer.QueryTimeSample(prop.path, time)
+                    prop_data.append(value.path)
+
+    return prop_data
```

Comment:
This may be nicer with less nesting.

```suggestion
    prop_data = []
    for prop in prim.properties:
        if not isinstance(prop, Sdf.AttributeSpec):
            continue
            
        if prop.typeName != Sdf.ValueTypeNames.Asset:
            continue
            
        default_val = prop.default
        if default_val:
            if hasattr(default_val, "__iter__"):
                prop_data.extend(default_val)
            else:
                prop_data.append(default_val.path)
        else:
            time_samples = layer.ListTimeSamplesForPath(prop.path)
            for time in time_samples:
                value = layer.QueryTimeSample(prop.path, time)
                prop_data.append(value.path)

    return prop_data
```

Also, `prop_data` may be refactored to what it actually contains, maybe `asset_paths`? or `property_asset_paths`?